metaflow/runner/subprocess_manager.py

```diff
@@ -81,8 +125,14 @@ def run_command(
         """
 
         command_obj = CommandManager(command, env, cwd)
-        pid = command_obj.run(show_output=show_output)
+        pid = command_obj.run(show_output=show_output, wait=False)
+
         self.commands[pid] = command_obj
+
+        command_obj.process.wait()
+        command_obj.stdout_thread.join()
+        command_obj.stderr_thread.join()
```

Comment:
Not happy about this, but its needed in order to populate `self.commands`. `wait` was already taken with an async impl. Not sure if i can change interfaces and potentially break any code that depends on this. Behaviour for any other code won't change due to the new defaulted param `wait` in `run`.